File: SistFacturacion/src/main/java/paquete/Televisor.java
package paquete;

public class Televisor extends  Electrodomestico{
private double tamaño;
private boolean sintonizador;

    public Televisor(String opcImportado, Consumo consumo, double tamaño, boolean sintonizador){
        super(opcImportado, consumo);
        this.tamaño = tamaño;
        this.sintonizador = sintonizador;
    }

    public double getTamaño() {
        return tamaño;
    }

    public void setTamaño(double tamaño) {
        this.tamaño = tamaño;
    }

    public boolean isSintonizador() {
        return sintonizador;
    }

    public void setSintonizador(boolean sintonizador) {
        this.sintonizador = sintonizador;
    }

    public  double adicionarPulgadas(){
        if(getTamaño()>40){
           return getTotalBase()*0.3;
        }
        return 0;
    }

    public double adicionarSintetizdor(){
        if(isSintonizador()){
           return 250000;
        }
        return 0;
    }

@Override
    public  double calcularPrecioBase(){
        return getTotalBase() +adicionarPulgadas()+adicionarSintetizdor();
    }
}

File: Televisor.js
let Electrodomesticos = require('./Electrodomesticos');

module.exports = class Televisor extends Electrodomesticos{
    constructor(opcProcedencia, consumo, tamaño, sintotizador){
        super(opcProcedencia, consumo);
        this.tamaño = tamaño;
        this.sintotizador = sintotizador;
    }

    adicionarPulgadas(){
         let resul = (this.tamaño > 40)?this.precio*0.3:0;
        return resul;
    }

    adicionarSintetizador(){
        let resul2 = (this.sintotizador)?250000:0;
        return resul2;
    }


    calcularPrecioBase(){
        super.calcularPrecioBase();
        return this.precio+this.adicionarSintetizador()+ this.adicionarPulgadas();
    }
}


